pkg/syncer/syncer_process_test.go

```diff
@@ -0,0 +1,859 @@
+					"state.internal.workloads.kcp.dev/us-west1": "Sync",
+				},
+					map[string]string{
+						"kcp.dev/namespace-locator": "{\"logical-cluster\":\"root:org:ws\",\"namespace\":\"test\"}",
+					}),
+				deployment("theDeployment", "kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "", map[string]string{
+					"state.internal.workloads.kcp.dev/us-west1": "Sync",
+				}, nil, []string{"workloads.kcp.dev/syncer-us-west1"}),
+			},
+			fromResource: deployment("theDeployment", "test", "root:org:ws",
+				map[string]string{"state.internal.workloads.kcp.dev/us-west1": "Sync"},
+				map[string]string{
+					"deletion.internal.workloads.kcp.dev/us-west1": time.Now().Format(time.RFC3339),
+					"finalizers.workloads.kcp.dev/us-west1":        "another-controller-finalizer",
+				},
+				[]string{"workloads.kcp.dev/syncer-us-west1"}),
+			resourceToProcessLogicalClusterName: "root:org:ws",
+			resourceToProcessName:               "theDeployment",
+			workloadClusterName:                 "us-west1",
+			advancedSchedulingEnabled:           true,
+
+			expectError:         false,
+			expectActionsOnFrom: []clienttesting.Action{},
+			expectActionsOnTo: []clienttesting.Action{
+				createNamespaceAction(
+					"",
+					changeUnstructured(
+						toUnstructured(t, namespace("kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "",
+							map[string]string{
+								"state.internal.workloads.kcp.dev/us-west1": "Sync",
+							},
+							map[string]string{
+								"kcp.dev/namespace-locator": "{\"logical-cluster\":\"root:org:ws\",\"namespace\":\"test\"}",
+							})),
+						removeNilOrEmptyFields,
+					),
+				),
+				patchDeploymentAction(
+					"theDeployment",
+					"kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973",
+					types.ApplyPatchType,
+					toJson(t,
+						changeUnstructured(
+							toUnstructured(t, deployment("theDeployment", "kcp0124d7647eb6a00b1fcb6f2252201601634989dd79deb7375c373973", "", map[string]string{
+								"state.internal.workloads.kcp.dev/us-west1": "Sync",
+							}, map[string]string{
+								"deletion.internal.workloads.kcp.dev/us-west1": time.Now().Format(time.RFC3339),
+								"finalizers.workloads.kcp.dev/us-west1":        "another-controller-finalizer",
+							}, nil)),
+							// TODO(jmprusi): Those next changes do "nothing", it's just for the test to pass
+							//                as the test expects some null fields to be there...
+							setNestedField(nil, "spec", "selector"),
+							setNestedField(map[string]interface{}{}, "spec", "strategy"),
+							setNestedField(map[string]interface{}{
+								"metadata": map[string]interface{}{
+									"creationTimestamp": nil,
+								},
+								"spec": map[string]interface{}{
+									"containers": nil,
+								},
+							}, "spec", "template"),
+							setNestedField(map[string]interface{}{}, "status"),
+						),
+					),
+				),
+			},
+		},
+		"StatusSyncer with AdvancedSchedulingFeature": {
```

Comment:
StatusSyncer what? Be a little more descriptive. Is this about updating a downstream object? Or creating, or deleting?

Reply:
done